experimenter/experimenter/features/manifests/nimbus_fml_loader.py

```diff
@@ -22,75 +24,75 @@ def __init__(self, application: str, channel: str):
     @classmethod
     @lru_cache
     def create_loader(cls, application: str, channel: str):
+        """Application and channel should stay the same for each iteration of the 
+        FML loader.
+        """
         return cls(application, channel)
 
-    def file_path(self):
+    def file_path(self, version: NimbusFeatureVersion = None):
         """Get path to release feature manifest from experimenter (local)."""
+
         if self.application is not None:
-            path = Path(self.MANIFEST_PATH, self.application, f"{self.channel}.fml.yaml")
+            path = Path(self.MANIFEST_PATH, self.application)
+            if version:
+                path /= f"v{version}"
+            path /= f"{self.channel}.fml.yaml"
+
             if Path.exists(path):
                 return path
+            else:
+                logger.error(f"Nimbus FML Loader: Invalid manifest path: {path}")
+                return None
         else:
             logger.error(
                 "Nimbus FML Loader: Invalid application. Failed to get local "
                 "feature manifest."
             )
         return None
 
-    # Todo: Add versioning https://mozilla-hub.atlassian.net/browse/EXP-3875
-    def _get_fml_clients(self) -> list[FmlClient]:
-        clients = []
-        if file_path := self.file_path():
-            client = self.fml_client(str(file_path), self.channel)
-            if client is not None:
-                clients.append(client)
-            else:
-                logger.error(
-                    "Nimbus FML Loader: "
-                    f"Failed to create FML clients for file path {file_path} and "
-                    f"channel {self.channel}."
-                )
-        return clients
-
     @lru_cache  # noqa: B019
-    def fml_client(self, path: str, channel: str) -> FmlClient:
-        return FmlClient(path, channel)
+    def fml_client(self, version: Optional[NimbusFeatureVersion] = None) -> FmlClient:
+        """There is a single FmlClient for each combination of application, app version,
+        and channel.
+        """
+        return FmlClient(
+            str(self.file_path(version)),
+            self.channel,
+        )
 
-    def _get_inspectors(
-        self, clients: list[FmlClient], feature_id: str
-    ) -> list[FmlFeatureInspector]:
-        return [
-            inspector
-            for inspector in (
-                client.get_feature_inspector(feature_id) for client in clients
-            )
-            if inspector is not None
-        ]
+    @lru_cache  # noqa: B019
+    def feature_inspector(
+        self, client: FmlClient, feature_id: str
+    ) -> FmlFeatureInspector:
+        """There is a single FmlFeatureInspector for each feature."""
+        return client.get_feature_inspector(feature_id)
```

Comment:
We don't need to cache this.

[Here](https://github.com/mozilla/application-services/blob/8e0faa93c56caaaf7309dd986d904aac99a47e9b/components/support/nimbus-fml/src/client/inspector.rs#L14-L21
) is the rust code:
```rust
impl FmlClient {
    pub fn get_feature_inspector(&self, id: String) -> Option<Arc<FmlFeatureInspector>> {
        _ = self.manifest.find_feature(&id)?;
        Some(Arc::new(FmlFeatureInspector::new(
            self.manifest.clone(),
            id,
        )))
    }
}
```

This is just copying a string and a two atomic integer adds. Plus whatever wrapping code there is. This is not worth caching.

Reply:
I hadn't rebased this yet, it's been removed :D